### ISSUE TITLE
False-positive error on overriding attribute of base class with rtype-annotated property 

### ISSUE THREAD

**Issue by sizmailov** [python/mypy#9134]
*(bug report)*

Annotation of property of derived class with `-> Any` makes `mypy` complain about incompatibility with base class. The exact rtype doesn't matter, the error is still the same. The MRE is a simplified version of `stubgen` output. 

**MRE:**

```python
from typing import Any

class A:
    doc = ""

class B(A):
    # remove "-> Any" to make mypy happy
    @property
    def doc(self) -> Any:
        return ""
```

**Expected:** no errors reported

**Actual:**

```
foo.py:11: error: Signature of "doc" incompatible with supertype "A"
Found 1 error in 1 file (checked 1 source file)
```
**command line command:** `mypy foo.py`

**version info:**
mypy: 0.790+dev.ffdbeb3d47ccb2d9691b36993be0a18e0718d997
python: 3.7.3 

P.S.: The error message of `Signature of "doc" incompatible with supertype "A"` should be more descriptive. It's good enough for simple cases, but rather cryptic for more involved examples. (I realize it might be not so trivial to propagate the root source of incompatibility and represent it nicely)

**Comment by JukkaL** [comment]
The error message is kind of reasonable, since `doc` is mutable in `A` but not in `B`. If you don't have an annotation, mypy treats the code as untyped and omits various checks.

> The error message of Signature of "doc" incompatible with supertype "A" should be more descriptive

I think that mypy sometimes produces a better message, but not in this case. Would you like to create a separate issue about the better error message?

**Comment by sizmailov** [comment]
> The error message is kind of reasonable, since doc is mutable in A but not in B.

If it so, the next snippet should pass the check, right? 

```python
from typing import Any


class A:
    doc = ""


class B(A):
    @property
    def doc(self) -> Any:
        return ""

    @doc.setter
    def doc(self, value: str) -> None:
        print("set")

    @doc.deleter
    def doc(self) -> None:
        print("del")

```

In fact `mypy` reports the same error: 

```
example.py:9: error: Signature of "doc" incompatible with supertype "A"
example.py:10: error: Signature of "doc" incompatible with supertype "A"
Found 2 errors in 1 file (checked 1 source file)
```

At the moment I'm not sure what makes `mypy` complain about this snippet to file another issue.

**Comment by sizmailov** [comment]
Apparently `@property` decorator is treated as black box and effectively `() -> Any` gets compared to `str`.

Another form of overriding value-attribute with `property` triggers another false-positive error: 

```python
class B(A):
    doc = property(...)
# Incompatible types in assignment (expression has type "property", base class "A" defined the type as "str")
```

Same applies to custom [descriptors](https://docs.python.org/3/howto/descriptor.html#descriptor-example) as well:
```python
class B(A):
    doc = RevealAccess(10, 'var "x"')
# Incompatible types in assignment (expression has type "RevealAccess", base class "A" defined the type as "str")
```

**Comment by ilevkivskyi** [comment]
Now mypy gives a nice error
```
test.py:8: error: Cannot override writeable attribute with read-only property  [override]
```
and passes when setter is added.
